### PR TITLE
Use update function for Hooks.state setter

### LINKED ISSUE #26: Developer Experience: Stale `set` functions from Hooks
After upgrading to latest brisk - we experienced a couple regressions in our functionality:
- Slider control was no longer updating (#342)
- Scroll bounce was no longer working (#346)

These occurred after the update to the immutable hooks API: https://github.com/revery-ui/revery/commit/eda0fe65a38aa3a3793e00d7750478a5df51c5da

The issue was a problem in our code - we had long-lived closures that held on to `set` functions past re-renders. For #346 , for example, a fix was to switch to a shorter-lived `effect` hook. 

However, it wasn't very diagnosable - we just started seeing unexpected behavior when we used the `set` functions in this way.  It'd be helpful if we could at least detect this case - that we're calling a stale `set` and throw a descriptive error - like how the React devtools warn. 

A message like `ERROR: Stale setter used for reducer hook; check that you aren't holding a setter in a long-lived function` would be helpful to catch the error close to the source.

### PR DESCRIPTION
**Pull request by glennsl**
As discussed on the RNU Discord. Fixes #26 I believe.

The examples fail to build, for reasons unrelated to this. They seemed pretty outdated, with lots of lockfile updates, but I hope I got the changes there right anyway.

### CODE REVIEW

**Comment by wokalski** [comment]
Awesome, thanks!

File: lib/Hooks.re [hunk 1]
```diff
@@ -116,25 +110,10 @@ module State = {
...
-          which enforces handling of race conditions.
-          ",
-          backtrace,
-        );
-      } else {
-        setState(nextState, stateContainer);
-        onStateDidChange();
-      };
+    let setter = updater => {
+      setState(updater(stateContainer.currentValue), stateContainer);
```
Comment (wokalski) [inline lib/Hooks.re]: I think the updater should always pass `nextValue`.

Comment (glennsl) [reply lib/Hooks.re]: Yeah, that makes sense. Done.

**Comment by wokalski** [comment]
CI is fixed on master, merging!

**Pull request merged**